packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -952,10 +952,24 @@ export class JobService {
     return jobEntity;
   }
 
-  public async requestToCancelJob(userId: number, id: number): Promise<void> {
-    const jobEntity = await this.jobRepository.findOneByIdAndUserId(id, userId);
+  public async requestToCancelJob(
+    userId: number,
+    id: number,
```

Comment:
It is not a good practice to use id input para as chainid and jobid. I suggest to have two methods:
- requestToCancelJobById(userId: number, jobId: number)
- requestToCancelJobByAddress(userId: number, chainId: number, escrowAddress: string)

Then a third method being called from previous ones:
- requestToCancelJob(job: JobEntity)

Reply:
I will change it